### PR TITLE
Bugfix #3989

### PR DESCRIPTION
**Pull request by Axel32019**
enable collision detection during field work
- in AIDriver.lua enable collision detection by default in init(vehicle)
- in FieldworkAIDriver.lua disabled self:disableCollisionDetection() in changeToFieldwork()
- in TrafficCollision.lua getStatus(dt) raise isInTraffic for colliding vehicles which do not move
- in turn.lua added collision detection from AIDriver based modes

### CODE REVIEW

**Comment by pvaiko** [comment]
How is #3995 doing? Why fieldworkers won't stop due to false alarms colliding with the implement/trailer?

**Comment by Axel32019** [comment]
Problem: 
own connected implements and worktools are detected as collision with other object
Root caus:
CollisionDetector:addToIgnoreList(object) does not add the connected implements and worktools to the own ignorelist, as this information is not available when CollisionDetector is initialized with onLoad(savegame) by the call courseplay:setAIDriver(self, self.cp.mode)
Solution:
Moved the intialisation of CollisionDetector to AIDriver:beforeStart()

File: turn.lua [hunk 1]
```diff
@@ -693,6 +693,20 @@ function courseplay:turn(vehicle, dt, turnContext)
 	----------------------------------------------------------
 	--allowedToDrive = allowedToDrive and not courseplay:needToWaitForTools(vehicle)
 
+
+	if vehicle.cp.drivingMode:get() == DrivingModeSetting.DRIVING_MODE_AIDRIVER then
+		-- in AIDriver based modes do not call / use legacy collision code
+		if vehicle.cp.driver.collisionDetector then
```
Comment (pvaiko) [inline turn.lua]: Can we wrap this section into a AIDriver:isInTraffic() function?

File: FieldworkAIDriver.lua [hunk 2]
```diff
@@ -339,7 +339,7 @@ end
 
 function FieldworkAIDriver:changeToFieldwork()
 	self:debug('change to fieldwork')
-	self:disableCollisionDetection()
+	-- self:disableCollisionDetection()
```
Comment (pvaiko) [inline FieldworkAIDriver.lua]: no dead code please

File: turn.lua (hunk 1 above)
Comment (Axel32019) [reply turn.lua]: Done.

File: FieldworkAIDriver.lua (hunk 2 above)
Comment (Axel32019) [reply FieldworkAIDriver.lua]: Done.

File: AIDriver.lua [hunk 3]
```diff
@@ -152,7 +152,8 @@ function AIDriver:init(vehicle)
 	-- if someone wants to stop by calling hold()
 	self.allowedToDrive = true
 	self.collisionDetectionEnabled = true
-	self.collisionDetector = CollisionDetector(self.vehicle)
+	-- self.collisionDetector = CollisionDetector(self.vehicle)
```
[the remark this answers is not in the document]
Comment (Axel32019) [reply AIDriver.lua]: Done.

File: turn.lua (hunk 1 above)
Comment (pvaiko) [reply turn.lua]: Ok, but now it is the same as detectCollision() except the info text is not set and speed is regulated. Is this what we want? Or could just detectCollision() return a bool?

Comment (Axel32019) [reply turn.lua]: You are right. The difference will be the TRAFFIC message. I will use detectCollision() with return value self.allowedToDrive - OK?

Comment (pvaiko) [reply turn.lua]: ok

**Pull request merged**